fdbclient/VersionedMap.h

```diff
@@ -114,30 +114,51 @@ namespace PTreeImpl {
 		return contains(p->child(!less, at), at, x);
 	}
 
+	// TODO: Remove the number of invocations of operator<, and replace with something closer to memcmp.
```

Comment:
`bool operator<(const StringRef&, const StringRef&)` is implemented using memcmp already, is that not close enough? Is the goal to not lose the extra information memcmp provides? Then we wouldn't need to call operator< twice to confirm `x == p->data` below.

Btw https://github.com/apple/foundationdb/pull/2875 seems related.

Reply:
The goal is to avoid losing the extra information.  I'm working on a second PR that adds a compare operator to everything that's stored by these data structures (and remove the call to confirm ==), but that's a more invasive change.